middleware/forwarded.go

```diff
@@ -0,0 +1,42 @@
+package middleware
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+)
+
+// extractHost returns the Host IP address without any port information
+func extractHost(address string) string {
+	hostIP := net.ParseIP(address)
+	if hostIP != nil {
+		return hostIP.String()
+	}
+	var err error
+	hostStr, _, err := net.SplitHostPort(address)
+	if err != nil {
+		// Invalid IP address, just return it so it shows up in the logs
+		return address
+	}
+	return hostStr
+}
+
+// GetSource extracts the X-FORWARDED-FOR header from the given HTTP request
+// and returns a string with it and the remote address
```

Comment:
For such a long comment this is inexact.  How about "returns any source addesses we can find in the request, comma-separated"?  Anyone interested in how it is done can read the source.

Reply:
Changed as suggested.